File: src/main/resources/scripts/Examples/JavaScript/Move_Machine.js
/**
 * Move the default nozzle in a 10mm square.
 * Demonstrates safely submitting machine tasks and using OpenPnP
 * Java types in scripts. Also shows how to define a utility function
 * in an external script file and call it.
 */

// Load Examples/Utility.js functions into our scope. This will define the
// task function.
load(scripting.getScriptsDirectory().toString() + '/Examples/JavaScript/Utility.js');

// Import some OpenPnP classes we'll use
var imports = new JavaImporter(org.openpnp.model, org.openpnp.util);

// Using the imports from above, do some work.
with (imports) {
	task(function() {
		var nozzle = machine.defaultHead.defaultNozzle;
		var location = nozzle.location;

		// Move 10mm right
		location = location.add(new Location(LengthUnit.Millimeters, 10, 0, 0, 0));
		nozzle.moveTo(location);

		// Move 10mm up
		location = location.add(new Location(LengthUnit.Millimeters, 0, 10, 0, 0));
		nozzle.moveTo(location);

		// Move 10mm left
		location = location.add(new Location(LengthUnit.Millimeters, -10, 0, 0, 0));
		nozzle.moveTo(location);

		// Move 10mm down
		location = location.add(new Location(LengthUnit.Millimeters, 0, -10, 0, 0));
		nozzle.moveTo(location);
	});
}

File: src/main/resources/scripts/Examples/Python/move_machine.py
from __future__ import absolute_import, division

from org.openpnp.model import LengthUnit, Location

# submitUiMachineTask should be used for all code that interacts
# with the machine. It guarantees that operations happen in the
# correct order, and that the user is presented with a dialog
# if there is an error.
from org.openpnp.util.UiUtils import submitUiMachineTask

def main():
    submitUiMachineTask(move_nozzle)

def print_location(location):
    print('Location: {}'.format(location.toString()))


def move_nozzle():
    nozzle = machine.defaultHead.defaultNozzle
    location = nozzle.location
    print_location(location)

    # Move 10mm right
    location = location.add(Location(LengthUnit.Millimeters, 10, 0, 0, 0))
    print_location(location)
    nozzle.moveTo(location)

    # Move 10mm up
    location = location.add(Location(LengthUnit.Millimeters, 0, 10, 0, 0))
    print_location(location)
    nozzle.moveTo(location)

    # Move 10mm left
    location = location.add(Location(LengthUnit.Millimeters, -10, 0, 0, 0))
    print_location(location)
    nozzle.moveTo(location)

    # Move 10mm down
    location = location.add(Location(LengthUnit.Millimeters, 0, -10, 0, 0))
    print_location(location)
    nozzle.moveTo(location)

main()
